usr/sbin/p11sak/p11sak.c

```diff
@@ -2675,6 +2933,9 @@ int main(int argc, char *argv[])
         goto done;
     }
 
+    /* free */
+    confignode_deepfree(cfg);
```

Comment:
move `confignode_deepfree(cfg);` to after the `done: `label, you need to free the config also when one of the above steps has failed.  Maybe check cfg for being NULL before calling c`onfignode_deepfree()` , if `confignode_deepfree()` not already ignores an NULL config. You may reach the done label even before it has been parsed, so it might be NULL....

Reply:
`confignode_deepfree` is safe to call with `NULL`.  So just unconditionally call it after `done:`.

Reply:
Done